actionview/lib/action_view/helpers/form_options_helper.rb

```diff
@@ -707,6 +707,24 @@ def collection_radio_buttons(object, method, collection, value_method, text_meth
       #   collection_check_boxes(:post, :author_ids, Author.all, :id, :name_with_initial) do |b|
       #      b.label(:"data-value" => b.value) { b.check_box + b.text }
       #   end
+      #
+      # ==== Gotcha
+      #
+      # The HTML specification says when nothing is select on a collection of radio buttons
+      # web browsers do not send any value to server.
+      # Unfortunately this introduces a gotcha:
+      # if a +User+ model has a +category_ids+ field, and in the form none category is selected no +category_ids+ parameter is sent. So,
+      # any mass-assignment idiom like
```

Comment:
For example, if we have a +User+ model with +category_ids+ field and we have the following code in our update action:

  @user.update(params[:user])

If no +category_ids+ are selected then we can safely assume this field will not be updated.

This is possible thanks to a hidden field generated by the helper method for every collection of radio buttons. This hidden field is given the same field name as the radio buttons with a blank value.

In the rare case you don't want this hidden field, you can pass the <tt>include_hidden: false</tt> option to the helper method.